cardano_node_tests/tests/kes.py

```diff
@@ -0,0 +1,56 @@
+"""Functionality for KES key used in multiple tests modules."""
+import logging
+
+LOGGER = logging.getLogger(__name__)
+
+
+# valid scenarios when we are testing the kes-period-info cli command
+class KesScenarios:
+    ALL_VALID = "all_valid"
+    ALL_INVALID = "all_invalid"
+    INVALID_COUNTERS = "invalid_counters"
+    INVALID_KES_PERIOD = "invalid_kes_period"
+
+
+def check_kes_period_info_result(kes_output: dict, expected_scenario: str):
+    """Check output from kes-period-info command."""
+    output_scenario = None
+
+    # expected scenario: all_valid
+    if kes_output.get("valid_counters") and kes_output.get("valid_kes_period"):
+        # check metrics, metrics dict are only returned by the command if everything is valid
+        metrics = kes_output.get("metrics") or {}
+        node_state_counter = metrics.get("qKesNodeStateOperationalCertificateNumber") or 0
+        on_disk_counter = metrics.get("qKesOnDiskOperationalCertificateNumber") or 0
+        start_kes_period = metrics.get("qKesStartKesInterval") or 0
+        end_kes_period = metrics.get("qKesEndKesInterval") or 0
+        current_kes_period = metrics.get("qKesCurrentKesPeriod") or 0
+
+        metrics_to_compare = [
+            node_state_counter,
+            on_disk_counter,
+            start_kes_period,
+            end_kes_period,
+            current_kes_period,
+        ]
+
+        # check if the metrics match with the expected scenario
+        if (
+            all(isinstance(value, int) for value in metrics_to_compare)
+            and node_state_counter <= on_disk_counter
+            and start_kes_period <= current_kes_period <= end_kes_period
+        ):
+            output_scenario = KesScenarios.ALL_VALID
+    # expected scenario: all_invalid
+    elif not kes_output.get("valid_counters") and not kes_output.get("valid_kes_period"):
```

Comment:
Optional:
`not something and not something_else`
is customary abbreviated as
`not (something or something_else)`